cylc/flow/scripts/cylc.py

```diff
@@ -88,7 +88,7 @@ def get_version(long=False):
     from pathlib import Path
     version = f"{__version__}"
     if long:
-        version += f" ({Path(sys.executable).parent.parent})"
+        version += f" ({Path(sys.argv[0])})"
```

Comment:
I'm a bit worried that this might not produce the desired result in all cases, e.g:

```console
$ python -c 'import sys; print(sys.argv[0])'
-c
```

Reply:
Produces the right path in all testing I've done. Not sure how it could not, as it will always be run via the `cylc` script?

Reply:
Yes, but the Cylc script itself is called by a wrapper script that is auto-generated from the entry points.

E.G, try:

```console
$ mamba activate <env>
$ cat $(which cylc)
```

This script may differ between different Mamba/Conda client implementations (incl conda-pack) as well as venv, virtualenv and native Python/pip.

Reply:
How about using `__file__`?

Reply:
Granted, it won't give the same result as `command -v cylc` if using a wrapper script; but it will still give the ultimate `cylc` executable path (tested).

I've tested my solution on GH Actions Ubuntu and MacOS 14 runners which aren't using conda, and get the right results.

> How about using `__file__`?

https://github.com/cylc/cylc-flow/blob/5b215901e336f4f8b64738fca3402e4548eef035/cylc/flow/scripts/cylc.py#L83-L86

Reply:
From testing, only the `-c` argument to `python` appears to be able to get `sys.argv[0]` to give the wrong answer.